### PR TITLE
adding conformance tests

### PR DESCRIPTION
**Pull request by FriedrichWilken**
First iteration of sonobuoy tests. Just run ``` ./hack/test_sonobuoy.sh ``` It should set up the whole test env and run the sonobuoy quick test. 

In future iterations, we could add more tests and run them via GH Actions.

### CODE REVIEW

File: hack/setup_sonobuoy.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  then
+    echo "kind could not be found"
+    exit
+  fi
+}
+
+function setup_sonobuoy() {
+  if [ -f ./sonobuoy/sonobuoy ]; then
+    echo "Found Sonobuoy, skipping setup."
+  else
```
Comment (knabben) [inline hack/setup_sonobuoy.sh:28]:
``` if ... then
      echo "Found Sonobuoy, skipping setup."
      exit 1
   fi
```

File: hack/setup_sonobuoy.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    architecture=$(dpkg --print-architecture)
+    url="https://github.com/vmware-tanzu/sonobuoy/releases/download/v0.55.0/sonobuoy_0.55.0_"$os"_"$architecture".tar.gz"
+    wget $url -O ./sonobuoy.tar.gz
+    tar -xf sonobuoy.tar.gz -C ./sonobuoy
+    rm sonobuoy.tar.gz
+  fi
+}
+
+function setup_kind() {
+  if kind get clusters | grep -q kpng-proxy; then
```
Comment (knabben) [inline hack/setup_sonobuoy.sh:40]: can be a good issue for kind the creation of `kind get cluster --name kpng-proxy` attribute

File: hack/setup_sonobuoy.sh (hunk 1 above)
Comment (jayunit100) [reply hack/setup_sonobuoy.sh:28]: will do this in a follow on , lets merge and iterate to get this in before the mtng today

Comment (FriedrichWilken) [reply hack/setup_sonobuoy.sh:28]: okay, @jayunit100. @nehaLohia27 could you please fix this in your further work on sonobuoy?

File: hack/setup_sonobuoy.sh (hunk 2 above)
Comment (FriedrichWilken) [reply hack/setup_sonobuoy.sh:40]: @nehaLohia27 again, could you please fix this in your further work on sonobuoy? 🙂

Comment (FriedrichWilken) [reply hack/setup_sonobuoy.sh:40]: good point

File: hack/setup_sonobuoy.sh (hunk 1 above)
Comment (jayunit100) [reply hack/setup_sonobuoy.sh:28]: yup merged making pr now

Comment (jayunit100) [reply hack/setup_sonobuoy.sh:28]: ah wait i did e 1 and that fails for other reasons bc i have sonobuoy already

**Pull request merged**